lib/services/provisioning.go

```diff
@@ -124,28 +124,27 @@ func MarshalProvisionToken(t ProvisionToken, opts ...MarshalOption) ([]byte, err
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	type token1 interface {
-		V1() *ProvisionTokenV1
-	}
-	type token2 interface {
-		V2() *ProvisionTokenV2
-	}
 
-	version := cfg.GetVersion()
-	switch version {
-	case V1:
-		v, ok := t.(token1)
-		if !ok {
-			return nil, trace.BadParameter("don't know how to marshal %v", V1)
+	version := t.GetVersion()
+	switch provisionToken := t.(type) {
+	case *types.ProvisionTokenV2:
+		// types.ProvisionTokenV1 doesn't implement ProvisionToken,
```

Comment:
nit: this is still confusing - why mention `types.ProvisionTokenV1` here since it can never make it inside this function and we're dealing with a V2 value that can convert to V1? I'd just say that this case is when V2 converts to V1